core/src/operator.rs

```diff
@@ -305,24 +305,32 @@ where
         Ok(())
     }
 
-    async fn is_profitable(&self, _withdrawal_idx: usize) -> Result<bool, BridgeError> {
-        // check that withdrawal_idx has the input_utxo.outpoint
-        // call is_profitable
-        // if is profitable, pay the withdrawal
-        // TODO: Implement this
-        Ok(true)
+    async fn is_profitable(&self, withdrawal_amount: u64) -> Result<bool, BridgeError> {
+        // Check if the withdrawal amount is within the acceptable range
+        if withdrawal_amount > self.config.bridge_amount_sats {
+            Ok(false)
+        } else {
+            // Calculate net profit after the withdrawal
+            let net_profit = self.config.bridge_amount_sats - withdrawal_amount;
+
+            // Check if the operator withdrawal fee is set and calculate profitability
+            match self.config.operator_withdrawal_fee_sats {
+                Some(fee) => Ok(fee < net_profit),
+                None => Err(BridgeError::OperatorWithdrawalFeeNotSet),
+            }
+        }
     }
 
     async fn new_withdrawal_sig(
         &self,
-        withdrawal_idx: usize,
+        _withdrawal_idx: usize,
         user_sig: schnorr::Signature,
         input_utxo: UTXO,
         output_txout: TxOut,
     ) -> Result<Option<Txid>, BridgeError> {
         // TODO: check that withdrawal_idx has the input_utxo.outpoint
 
-        if !self.is_profitable(withdrawal_idx).await? {
+        if !self.is_profitable(output_txout.value.to_sat()).await? {
```

Comment:
lets add input's dust value here as well. Let's not force users to create only dust values, they can create 1 BTC and request 11 BTC